### PR TITLE
Implement `Cortex.focus`

### PR DESCRIPTION
**Pull request by glittershark**
Implement a focus command, which filters the tests that run on changed files.

### CODE REVIEW

**Comment by rschmukler** [comment]
This is a fantastic bit of functionality. I haven't had a chance to review the implementation beyond the docs, but I am wondering if we could also support an `@focus true` annotation (or multiple annotations in a file) that would automatically register and de-register the focus. I believe that this would be the easiest to use and not requiring to remember an API (beyond a simple annotation) nor type out file names / paths.

**Comment by OvermindDL1** [comment]
Just remember that Elixir will puke warnings on unregistered attributes like `@focus true` would be.  That is why things like credo use comments.

**Comment by OvermindDL1** [comment]
Could always have them define something like:
```elixir
def __cortex__focus, do: true
```

**Comment by OvermindDL1** [comment]
Well unless `@focus` goes in the test file, if so just ignore me.  ^.^

**Comment by glittershark** [comment]
@rschmukler I don't feel like if `@focus true` existed it'd have to be mutually exclusive with this API - it'd be fine (and nice!) to have both.

File: lib/cortex/controller.ex [hunk 1]
```diff
@@ -7,52 +7,97 @@ defmodule Cortex.Controller do
...
   def pipeline(timeout \\ 5000) do
     GenServer.call(__MODULE__, :get_pipeline, timeout)
   end
 
+
   def run_all, do: GenServer.cast(__MODULE__, :run_all)
 
+
+  @doc """
+  Update the current focus of Cortex. Arguments are (currently) the same tag filters that would be
```
Comment (russmatney) [inline lib/cortex/controller.ex]: I'd drop (currently) here. Could see this getting out of date, and we might as well own it.

**Review by russmatney — APPROVED** [review]
One super minor doc tweak, but otherwise this looks legit – excited to start taking advantage of it. The auto-focus via a `@focus` tag mentioned by @rschmukler can come in a future PR.

File: lib/cortex.ex [hunk 2]
```diff
@@ -42,7 +42,41 @@ defmodule Cortex do
...
+    expression
+  - an integer, which will filter by line number for tests
+  - a keyword, which will pass through to `ExUnit.configure/1` unchanged
+
+  """
+  def focus(%Regex{} = re),
+    do: Cortex.Controller.set_focus(test: re)
+
+  def focus(string) when is_binary(string),
+    do: string |> Regex.compile() |> set_focus()
```
Comment (Camsbury) [inline lib/cortex.ex]: Should be `|> Cortex.Controller.set_focus()`?

**Review by Camsbury — DISMISSED** [review]
Noticed what could be a simple error. Otherwise this is awesome, and I need more exposure to Cortex/Stage to understand this better.

File: lib/cortex.ex (hunk 2 above)
Comment (glittershark) [reply lib/cortex.ex]: actually should be `|> focus()`. Good catch, though!

File: lib/cortex/controller.ex (hunk 1 above)
Comment (glittershark) [reply lib/cortex/controller.ex]: sure.

File: lib/cortex.ex (hunk 2 above)
Comment (Camsbury) [reply lib/cortex.ex]: Gotcha. Makes sense now.

**Comment by rschmukler** [comment]
To clarify my earlier comment, yes, I mean `@focus true` above a test definition.

**Pull request merged**